sql-plugin/src/main/scala/com/nvidia/spark/rapids/Plugin.scala

```diff
@@ -288,19 +289,21 @@ class RapidsExecutorPlugin extends ExecutorPlugin with Logging {
   override def onTaskFailed(failureReason: TaskFailedReason): Unit = {
     failureReason match {
       case ef: ExceptionFailure =>
-        val unrecoverableErrors = Seq("cudaErrorIllegalAddress", "cudaErrorLaunchTimeout",
-          "cudaErrorHardwareStackError", "cudaErrorIllegalInstruction",
-          "cudaErrorMisalignedAddress", "cudaErrorInvalidAddressSpace", "cudaErrorInvalidPc",
-          "cudaErrorLaunchFailure", "cudaErrorExternalDevice", "cudaErrorUnknown",
-          "cudaErrorECCUncorrectable")
-        if (unrecoverableErrors.exists(ef.description.contains(_)) ||
-          unrecoverableErrors.exists(ef.toErrorString.contains(_))) {
-          logError("Stopping the Executor based on exception being a fatal CUDA error: " +
-            s"${ef.toErrorString}")
-          System.exit(20)
+        ef.exception match {
+          case _: CudaFatalException =>
+            logError("Stopping the Executor based on exception being a fatal CUDA error: " +
+              s"${ef.toErrorString}")
+            System.exit(20)
+          case _: CudaException =>
+            logDebug(s"Executor onTaskFailed because of a non-fatal CUDA error: " +
+              s"${ef.toErrorString}")
+          case _: CudfException =>
+            logDebug(s"Executor onTaskFailed because of a CUDF error: ${ef.toErrorString}")
```

Comment:
My bad, ef.exception is an option so this needs to unpack it, e.g:
```suggestion
          case Some(_: CudaFatalException) =>
            logError("Stopping the Executor based on exception being a fatal CUDA error: " +
              s"${ef.toErrorString}")
            System.exit(20)
          case Some(_: CudaException) =>
            logDebug(s"Executor onTaskFailed because of a non-fatal CUDA error: " +
              s"${ef.toErrorString}")
          case Some(_: CudfException) =>
            logDebug(s"Executor onTaskFailed because of a CUDF error: ${ef.toErrorString}")
```